### PR TITLE
Add Gammapy overview docs page

### PR DESCRIPTION
**Pull request by cdeil**
This PR adds a first version of the Gammapy overview docs page.

The idea is to have all major concepts and classes mentioned, and to include many links.

The plan is to put this now quickly, and then to make Figures, polish the text, and make this overview really good in December, and to keep #2592 open. Writing a good overview page also requires many improvements to other parts of the docs, especially `tutorials/overview.ipynb`, so writing it has to be an iterative process to see what content fits best where.

Work in progress ... I'll try to help get #2621 in now, and then come back to this PR.

### CODE REVIEW

File: docs/overview.rst [hunk 1]
```diff
@@ -5,33 +5,105 @@
...
+tutorial notebook <notebooks/overview.html>`__ can be seen as an example-based
+Gammapy overview that parallels this more theoretical description.
+
+Introduction
+------------
+
+Gammapy is an open-source Python package for gamma-ray astronomy built on Numpy
+and Astropy. It is a prototype for the Cherenkov Telescope Array (CTA) science
+tools, and can also be used to analyse data from existing gamma-ray telescopes
+if their data is available in the standard FITS format.
```
Comment (registerrier) [inline docs/overview.rst]: Add DL3 FITS format and a link?

Comment (cdeil) [reply docs/overview.rst]: done

**Comment by cdeil** [comment]
@adonath or @registerrier - Can you please finish this to have a minimal viable page for v0.15? Thanks!

**Comment by cdeil** [comment]
I've finished a minimal `docs/overview.rst`, that is OK for v0.15.

I'll merge now without further review / discussion, because IMO we should instead focus on other parts that are not done at all or in very poor shape, e.g. `tutorials/overview.ipynb` or `tutorials/analysis_2.ipynb`, `tutorials/modeling.ipynb`, the changelog, ....

This is a first draft of `docs/overview.rst`, it should be improved of course in the coming weeks.

**Pull request merged**